src/dynamics/_common/track_tags.js

```diff
@@ -74,22 +82,27 @@ Scoped.define("module:TrackTags", [
              */
             _loadTrackTags: function() {
                 if (!this._dyn.get("tracktagssupport")) return;
-                var _flag = true;
-                Objs.iter(this._trackTags, function(subtitle, index) {
-                    var _trackTag = document.createElement("track");
-                    var _domEvent = this.auto_destroy(new DomEvents());
+
+                const prevTrackElements = this._video?.querySelectorAll('track');
+                if (prevTrackElements && prevTrackElements.length > 0) {
+                    prevTrackElements.forEach(tr => this._video.removeChild(tr));
+                }
+
+                Objs.iter(this._trackTags, function(trackTag, index) {
+                    const domEvent = this.auto_destroy(new DomEvents());
```

Comment:
@sambua wouldn't it be better if we initialized this once on the constructor?
```suggestion
```

Reply:
@ntshcalleia good point, will apply related changes 👍